content/guides/dapps/journal.md

```diff
@@ -0,0 +1,564 @@
+help you learn.
+
+In `my-journal-dapp`, navigate to `anchor/programs/journal/src.lib.rs`. There
+will already be a template code generated in this folder. Let's delete it and
+start from scratch so we can walk through each step.
+
+### Define your anchor program:
+
+```rust
+use anchor_lang::prelude::*;
+
+// This is your program's public key and it will update automatically when you build the project.
+declare_id!("7AGmMcgd1SjoMsCcXAAYwRgB9ihCyM8cZqjsUqriNRQt");
+
+#[program]
+pub mod journal {
+    use super::*;
+}
+```
+
+### Define your program state:
+
+The state is the data structure used to define the information you want to save
+to the account. Since Solana on-chain programs do not have storage, the data is
+stored in accounts that live on the blockchain.
+
+When using anchor, the `#[account]` attribute macro is used to define your
+program state.
+
+```rust
+#[account]
+pub struct JournalEntryState {
+    pub owner: Pubkey,
+    pub title: String,
+    pub message: String,
+}
+```
+
+For this journal aApp, we'll be storing the journal's owner, the title of each
+journal entry, and the message of each journal entry.
+
+### Create a journal entry:
+
+Now, let's add an instruction to this program that creates a new journal entry.
+To do this, we will update the `#[program]` code that we already defined earlier
+to include an instruction for `create_journal_entry`.
+
+When creating a journal entry, the user will need to provide the `title` and
+`message` of the journal entry. So we need to add those two variables as
+additional parameters.
+
+When calling this instruction, we want to save the `owner` of the account, the
+journal entry `title`, and the journal entry `message` to the account's
+`JournalEntryState`.
+
+```rust
+#[program]
+mod journal {
+    use super::*;
+
+    pub fn create_journal_entry(
+        ctx: Context<CreateEntry>,
+        title: String,
+        message: String,
+    ) -> Result<()> {
+        msg!("Journal Entry Created");
+        msg!("Title: {}", title);
+        msg!("Message: {}", message);
+
+        let journal_entry = &mut ctx.accounts.journal_entry;
+        journal_entry.owner = ctx.accounts.owner.key();
+        journal_entry.title = title;
+        journal_entry.message = message;
+        Ok(())
+    }
+}
+```
+
+For the anchor framework, every instruction takes a `Context` type as its first
+argument. The `Context` macro is used to define a struct that encapsulates
+accounts that will be passed to a given instruction. Therefore, each `Context`
+must have a specified type with respect to the instruction. In this case, we
+need to define a data structure for `CreateEntry`.
+
+```rust
+#[derive(Accounts)]
+#[instruction(title: String, message: String)]
+pub struct CreateEntry<'info> {
+    #[account(
+        init_if_needed,
+        seeds = [title.as_bytes(), owner.key().as_ref()],
+        bump,
+        payer = owner,
+        space = 8 + 32 + 4 + title.len() + 4 + message.len()
```

Comment:
Can you update this to use the `INIT_SPACE` macro for static sizes here and maybe add a brief description of this vs manually calculating the space?

Reference: https://www.anchor-lang.com/docs/space#the-init-space-macro